deepchecks/utils/abstracts/train_test_performace.py

```diff
@@ -0,0 +1,85 @@
+# ----------------------------------------------------------------------------
+# Copyright (C) 2021-2023 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+"""Module containing the train test performance check."""
+import abc
+import typing as t
+
+from typing_extensions import Self
+
+from deepchecks.core.check_utils.class_performance_utils import (
+    get_condition_class_performance_imbalance_ratio_less_than, get_condition_test_performance_greater_than,
+    get_condition_train_test_relative_degradation_less_than)
+from deepchecks.utils.strings import format_percent
+
+__all__ = ['TrainTestPerformanceAbstract']
+
+
+class TrainTestPerformanceAbstract:
+    """Base functionality for some train-test performance checks."""
+
+    @classmethod
+    @abc.abstractmethod
+    def _default_per_class_scorers(cls) -> t.Mapping[str, str]:
+        raise NotImplementedError()
```

Comment:
Why is this needed? 
In general the more prefered flow is to get the scorers list from context.get_scorers(self.scorers, use_avg_defaults=False) - as it was in the tabular sub package

Reply:
because `add_condition_class_performance_imbalance_ratio_less_than` needs access to the default scorers

```
def add_condition_class_performance_imbalance_ratio_less_than(
    self: Self,
    threshold: float = 0.3,
    score: t.Optional[str] = None
) -> Self:
    if score is None:
        score = next(iter(self._default_per_class_scorers()))

```

Reply:
wow that is some bad implementation.
Change the condition so it will be as in deepchecks/vision/checks/model_evaluation/class_performance.py
i.e the scorer name is received as input from the user

Reply:
oh, I see why it was done so, to not change the function signature, to not break backward compatibility

anyway, I do not see this as a problematic change/